java-to-zod-core/src/main/java/sh/ivan/zod/plugins/PluginParameters.java

```diff
@@ -0,0 +1,320 @@
+package sh.ivan.zod.plugins;
+
+import cz.habarta.typescript.generator.*;
+import java.io.Serializable;
+import java.util.Collections;
+import java.util.List;
+import lombok.Setter;
+import org.apache.maven.plugins.annotations.Parameter;
+import org.gradle.api.tasks.Input;
+import org.gradle.api.tasks.Optional;
+
+@Setter
+public class PluginParameters implements Serializable {
```

Comment:
We can generate the appropriate getters in this way:
```suggestion
@Getter(onMethod=@__({@Input}))
public class PluginParameters implements Serializable {
```
Not sure of the exact syntax. See Lombok docs: https://projectlombok.org/features/GetterSetter

Reply:
Do we actually need to implement Serializable too? I would also like to see this class extended by the Maven Mojo and the Gradle equivalent, so that the config properties are defined at the top level. I think that may eliminate the need for Serializable.

I guess to make this happen, we would need to extend both `AbstractMojo` and `DefaultTask` in `PluginParameters`. I think in that case, it makes more sense to name this `BasePlugin` or something.